src/components/BookLogic.js

```diff
@@ -22,12 +26,36 @@ const BookLogic = () => {
   };
 
   const deleteBook = (id) => {
-    dispatch(removeBook(id));
+    dispatch(removeBookFromApi(id));
   };
 
+  useEffect(() => {
+    dispatch(getBooks());
+  }, [dispatch]);
```

Comment:
- Optional
You can remove `dispatch` from the dependency array since it will remain same between re-renders.

Reply:
Thanks for the amazing review @muneebulrehman